ecr-login/api/factory.go

```diff
@@ -22,6 +22,7 @@ import (
 
 type ClientFactory interface {
 	NewClient(awsSession *session.Session, awsConfig *aws.Config) Client
+	NewClientWithCacheDir(awsSession *session.Session, awsConfig *aws.Config, cacheDir string) Client
```

Comment:
I think this will be very helpful if we want to add more configurations for the constructors in the future